pkg/lastpass/cli.go

```diff
@@ -36,24 +36,24 @@ func VerifyCliExistsOrDie() {
 
 // Login using lastpass-cli
 func Login(username string, password string) error {
-	// echo <PASSWORD> | LPASS_DISABLE_PINENTRY=1 lpass login --trust <USERNAME>
-	out, err := sh.NewSession().SetEnv("LPASS_DISABLE_PINENTRY", "1").Command("echo", password).Command("lpass", "login", "--trust", username).Output()
-	if err != nil || "" == string(out) {
-		// sometimes returns error: "Error: HTTP response code said error" even if the credentials are valid
-		return fmt.Errorf("verify credentials, unable to login: %s", err)
+	_, err := sh.Command("lpass", "status").Output()
+	log.Printf("Checking if already logged in")
+	if err != nil {
+		log.Printf("Doing login")
+		out, err := sh.NewSession().SetEnv("LPASS_DISABLE_PINENTRY", "1").Command("echo", password).Command("lpass", "login", "--trust", username).Output()
+		if err != nil || "" == string(out) {
+			// sometimes returns error: "Error: HTTP response code said error" even if the credentials are valid
+			return fmt.Errorf("verify credentials, unable to login: %s", err)
+		}
 	}
 	log.Printf("Succesfully logged in")
 	return nil
 }
 
 // Logout using lastpass-cli
 func Logout() {
-	// lpass logout --force
```

Comment:
i would probably keep the implementation and comment `lastpass.Logout()` in [here](https://github.com/edgelevel/lastpass-operator/blob/eab5d19984648424c23cf2fcbf475c39fc2cb59a/controllers/lastpass_controller.go#L81) - No point in keeping the method around otherwise

Reply:
I think it will be ok 👍  Thanks

Reply:
I'll look into it 😉